src/exchange/base_order_handler.cairo

```diff
@@ -186,55 +187,42 @@ mod BaseOrderHandler {
             starting_gas: u128,
             secondary_order_type: SecondaryOrderType
         ) -> ExecuteOrderParams {
-            let address_zero = 0.try_into().unwrap();
+            let data_store = self.data_store.read();
+
+            let order = order_store_utils::get(data_store, key);
+            let swap_path_markets = market_utils::get_swap_path_markets(
+                data_store, order.swap_path
+            );
+
             let execute_order_params_contract = ExecuteOrderParamsContracts {
-                data_store: self.data_store.read(),
+                data_store: data_store,
                 event_emitter: self.event_emitter.read(),
                 order_vault: self.order_vault.read(),
                 oracle: self.oracle.read(),
                 swap_handler: self.swap_handler.read(),
                 referral_storage: self.referral_storage.read(),
             };
-            let order = Order {
-                key: 0,
-                order_type: OrderType::MarketSwap(()),
-                decrease_position_swap_type: DecreasePositionSwapType::NoSwap(()),
-                account: address_zero,
-                receiver: address_zero,
-                callback_contract: address_zero,
-                ui_fee_receiver: address_zero,
-                market: address_zero,
-                initial_collateral_token: address_zero,
-                swap_path: Array32Trait::<ContractAddress>::span32(@ArrayTrait::new()),
-                size_delta_usd: 0,
-                initial_collateral_delta_amount: 0,
-                trigger_price: 0,
-                acceptable_price: 0,
-                execution_fee: 0,
-                callback_gas_limit: 0,
-                min_output_amount: 0,
-                updated_at_block: 0,
-                is_long: true,
-                should_unwrap_native_token: true,
-                is_frozen: true,
-            };
-            let market = Market {
-                market_token: address_zero,
-                index_token: address_zero,
-                long_token: address_zero,
-                short_token: address_zero,
-            };
+
+            let min_oracle_block_numbers = get_uncompacted_oracle_block_numbers(
+                oracle_params.compacted_min_oracle_block_numbers.span(), oracle_params.tokens.len()
+            );
+            let max_oracle_block_numbers = get_uncompacted_oracle_block_numbers(
+                oracle_params.compacted_max_oracle_block_numbers.span(), oracle_params.tokens.len()
+            );
+
+            let market = market_utils::get_enabled_market(data_store, order.market);
```

Comment:
is it ok if we don't check if `order.market != zero.address` before doing this?

Reply:
Better to check else I think it will revert inside `get_enabled_market`

Reply:
will do!